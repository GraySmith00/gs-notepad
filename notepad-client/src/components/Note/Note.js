import React from 'react';
import PropTypes from 'prop-types';
import { formatDate } from '../../lib/formatDate';

const Note = ({ text, createdAt, tag }) => {
  return (
    <div className="note">
      <h3>{text}</h3>
      <p>Time: {formatDate(createdAt)}</p>
      <p>Tag: {tag}</p>
    </div>
  );
};

Note.propTypes = {
  text: PropTypes.string.isRequired,
  createdAt: PropTypes.number.isRequired,
  tag: PropTypes.string.isRequired
};

export default Note;
