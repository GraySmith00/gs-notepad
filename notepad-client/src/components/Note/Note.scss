.note {
  display: flex;
  margin-bottom: 2rem;
  color: rgb(66, 71, 74);

  &__tag {
    font-style: italic;
    color: rgb(87, 95, 99);
    margin-top: 0.5rem;
    font-size: 1.4rem;
  }

  &__text {
    margin-left: 2rem;
  }
}
