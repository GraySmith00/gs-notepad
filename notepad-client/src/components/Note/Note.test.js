import React from 'react';
import { shallow } from 'enzyme';

import Note from './Note';

describe('Note component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Note text="new note" createdAt={1545951460092} tag="work" />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
