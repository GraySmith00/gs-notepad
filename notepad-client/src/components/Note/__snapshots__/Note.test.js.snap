// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Note
    createdAt={1545951460092}
    tag="work"
    text="new note"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          new note
        </h3>,
        <p>
          Time: 
          12/27/2018
        </p>,
        <p>
          Tag: 
          work
        </p>,
      ],
      "className": "note",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "new note",
        },
        "ref": null,
        "rendered": "new note",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Time: ",
            "12/27/2018",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Time: ",
          "12/27/2018",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Tag: ",
            "work",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Tag: ",
          "work",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            new note
          </h3>,
          <p>
            Time: 
            12/27/2018
          </p>,
          <p>
            Tag: 
            work
          </p>,
        ],
        "className": "note",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "new note",
          },
          "ref": null,
          "rendered": "new note",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Time: ",
              "12/27/2018",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Time: ",
            "12/27/2018",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Tag: ",
              "work",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Tag: ",
            "work",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
